crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2518,9 +2527,11 @@ impl<'db> TypeInferenceBuilder<'db> {
                 .map(Type::IntLiteral)
                 .unwrap_or_else(|| KnownClass::Int.to_instance(self.db)),
 
-            (Type::IntLiteral(_), Type::IntLiteral(_), ast::Operator::Div) => {
-                KnownClass::Float.to_instance(self.db)
-            }
+            (
+                Type::IntLiteral(_),
+                Type::IntLiteral(_) | Type::BooleanLiteral(_),
```

Comment:
I wouldn't include `BooleanLiteral` here explicitly. (If you did, it should also be included for left operand, as well as right operand, so as to handle e.g. `True / 2`.) But I'd instead allow this to be handled by the more generic recursive-call approach described below.